apis/sgv2-quarkus-common/src/main/java/io/stargate/sgv2/api/common/metrics/TenantRequestMetricsTagProvider.java

```diff
@@ -0,0 +1,75 @@
+package io.stargate.sgv2.api.common.metrics;
+
+import io.micrometer.core.instrument.Tag;
+import io.micrometer.core.instrument.Tags;
+import io.quarkus.micrometer.runtime.HttpServerMetricsTagsContributor;
+import io.stargate.sgv2.api.common.StargateRequestInfo;
+import io.stargate.sgv2.api.common.config.MetricsConfig;
+import io.vertx.core.http.HttpServerRequest;
+import jakarta.enterprise.context.ApplicationScoped;
+import jakarta.inject.Inject;
+import java.util.regex.Pattern;
+
+@ApplicationScoped
+public class TenantRequestMetricsTagProvider implements HttpServerMetricsTagsContributor {
+
+  // split pattern for the user agent, extract only first part of the agent
+  private static final Pattern USER_AGENT_SPLIT = Pattern.compile("[\\s/]");
+
+  // same as V1 io.stargate.core.metrics.StargateMetricConstants#UNKNOWN
+  private static final String UNKNOWN_VALUE = "unknown";
+
+  /** The configuration for metrics. */
+  private final MetricsConfig.TenantRequestCounterConfig config;
+
+  /** The request info bean. */
+  private final StargateRequestInfo requestInfo;
+
+  /** The tag for error being true, created only once. */
+  private final Tag errorTrue;
+
+  /** The tag for error being false, created only once. */
+  private final Tag errorFalse;
+
+  /** The tag for tenant being unknown, created only once. */
+  Tag tenantUnknown;
+
+  /** Default constructor. */
+  @Inject
+  public TenantRequestMetricsTagProvider(
+      StargateRequestInfo requestInfo, MetricsConfig metricsConfig) {
+    this.requestInfo = requestInfo;
+    this.config = metricsConfig.tenantRequestCounter();
+    errorTrue = Tag.of(config.errorTag(), "true");
+    errorFalse = Tag.of(config.errorTag(), "false");
+    tenantUnknown = Tag.of(config.tenantTag(), UNKNOWN_VALUE);
+  }
+
+  public Tags contribute(Context context) {
```

Comment:
Should this be `@Override`? (assuming it implements method for Provider interface)